tasks.toml

```diff
@@ -187,67 +187,217 @@ run = '''
 #!/bin/bash
 set -eu -o pipefail
 
-# Ensure on main branch
+# Ensure working directory is clean
+if [[ -n "$(git status --porcelain)" ]]; then
+  echo "ERROR: Working directory not clean"
+  git status
+  exit 1
+fi
+
 current_branch=$(git rev-parse --abbrev-ref HEAD)
 if [[ "$current_branch" != "main" ]]; then
   echo "ERROR: You must be on the 'main' branch (current: $current_branch)"
   exit 1
 fi
 
+# Fetch latest from origin to get current state
+git fetch origin
+
+# Checkout release branch (create if it doesn't exist)
+if git show-ref --verify --quiet refs/heads/release; then
+  # Branch exists locally, just checkout
+  git checkout release
+elif git show-ref --verify --quiet refs/remotes/origin/release; then
+  # Branch exists on remote, checkout and track
+  git checkout -b release origin/release
+else
+  # Branch doesn't exist, create it
+  git checkout -b release
+fi
+
+# Reset release branch to latest origin/main
+git reset --hard origin/main
+
+# Create RC via bump
+mise run bump
+
+# Checkout main and bump version
```

Comment:
I'm not sure I understand why the version is bumped on main and not on the `release` branch.

Also please deduplicate all the bumping logic, you can just call `mise bump` here